src/nrnpython/grids.cpp

```diff
@@ -25,7 +25,7 @@ extern double* _rxd_induced_currents_ecs;
 extern double* _rxd_induced_currents_scale;
 
 // Set dt, t pointers
-extern "C" void make_time_ptr(PyHocObject* my_dt_ptr, PyHocObject* my_t_ptr) {
+extern "C" NB_EXPORT void make_time_ptr(PyHocObject* my_dt_ptr, PyHocObject* my_t_ptr) {
```

Comment:
For general understanding, what is the rule around setting `NB_EXPORT`? It seems something we should not have to do ourselves...

Reply:
Ok, so when you publish a dynamic shared library there is some symbols exposed (you can see them with `nm`). Microsoft remove by default all symbols that are not used. So you cannot find them when you `dlopen` the library. So you have to tell the linker that you want to see them.

Here `extern "C"` means that you want a C linkage, because this is easier than `C++` linkage when you do some cross language symbols exports (avoid mangling and types).

So I suppose that all the `extern "C"` where the public API and I export them, because some people do `nrn_dll_sym` to find it.

Reply:
And no I was not able to use the `nanobind` file because the compilation is a bit weird `cmake` => `setup.py` => `cmake` and I was not able to easily fix inclusion of `nanobind` for the second python library.

First one `libnrnpython3.8.dll`, second `hoc.cpython-3.8.dll`.

Reply:
thanks for explanation Nico!

Reply:
https://www.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/ I live it here as an interesting blog post on the topic and CMake.